mono/eglib/glib.h

```diff
@@ -797,13 +797,17 @@ gunichar  *g_utf16_to_ucs4 (const gunichar2 *str, glong len, glong *items_read,
 gchar     *g_ucs4_to_utf8  (const gunichar *str, glong len, glong *items_read, glong *items_written, GError **err);
 gunichar2 *g_ucs4_to_utf16 (const gunichar *str, glong len, glong *items_read, glong *items_written, GError **err);
 
-#define u8to16(str) g_utf8_to_utf16(str, (glong)strlen(str), NULL, NULL, NULL)
+#define g_u16len monoeg_g_u16len
+gsize
+g_u16len (const gunichar2 *str);
```

Comment:
This needs to be renamed to `g_utf16_strlen`.

Reply:
We have both conventions and I thought about this a while and chose the shorter name. ok.

Reply:
`str` implies char*, `wcs` implies wchar_t* in ANSI C library. How about omiting `str` here? One of g_utf16len, g_utf16_length, g_utf16length? The other argument, though, is that `wcs` is obscure out side of Windows, `str` is all that is known, so prepend g_utf16_ and  then use exactly existing `str` function: g_utf16_strchr, g_utf16_strstr, g_utf16_strcmp, etc.